src/org/omegat/gui/dictionaries/DictionariesTextArea.java

```diff
@@ -257,8 +258,26 @@ protected void setFoundResult(final SourceTextEntry se, final List<DictionaryEnt
 
             displayedWords.add(de.getWord().toLowerCase());
             i++;
+
+            if (i % 30 == 0) {
+                String out = txt.toString();
+                if (textLength > 0) {
+                    moveCaretPosition(textLength);
+                    replaceSelection(out);
+                } else {
+                    setText(out);
+                }
+                txt = new StringBuilder();
+                textLength = out.length();
+            }
+        }
+        if (textLength > 0) {
+            moveCaretPosition(textLength);
+            replaceSelection(txt.toString());
+        } else {
+            setText(txt.toString());
```

Comment:
I would rather do something smarter, like working directly with the underlying `Document`: instead of doing a replacement you can call `Document#insertString` with the offset set to the length of the document.

How important is it to batch the appending (like you are doing here in chunks of 30)? If we can get away with not batching, I would prefer to remove it because it complicates the code.

Reply:
I'm not succeeded other than this change. 30 is a heuristics. when appending every line, things went worse.